src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +32,13 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .mapToObj(i -> (i % 2 != 0)
+                        ? Integer.valueOf(numbers.get(i) - 1)
+                        : Integer.valueOf(numbers.get(i)))
+                .filter(i -> (int) i % 2 != 0)
+                .mapToInt(i -> (int) i)
+                .average().getAsDouble();
```

Comment:
```suggestion
          .map(i -> i % 2 != 0 ? (numbers.get(i) - 1) : numbers.get(i))
          .filter(n -> n % 2 != 0)
          .average()
          .getAsDouble();
```

we can make it simpler